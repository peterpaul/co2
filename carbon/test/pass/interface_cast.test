
["stdio.h"] {
	int printf (char *format, ...);
}

interface Foo {
	int foobar ();
}

class Bar <Foo> {
	int foobar () return 5;
}

int main() {
	Foo foo = new Bar ();
	Bar bar = new Bar ();

	printf ("foobar = %d\n", ((Bar) foo).foobar());
	printf ("barfoo = %d\n", ((Foo) bar).foobar());

	return 0;
}
