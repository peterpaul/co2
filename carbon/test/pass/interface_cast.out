foobar = 5
barfoo = 5
